tools/lkl/lib/virtio_net_tap.c

```diff
@@ -122,5 +196,24 @@ struct lkl_netdev *lkl_netdev_tap_create(const char *ifname)
 		return NULL;
 	}
 
+	#ifdef CONFIG_EVENTFD
+	/* eventfd might be hijacked, so we can't use it */
+	nd->dev.ops = &tap_net_ops_uncloseable;
+	nd->eventfd = -1;
+
+	#else  /* CONFIG_EVENTFD */
+	/* eventfd can't be hijacked, so we can use it */
+	nd->dev.ops = &tap_net_ops_closeable;
+	nd->eventfd = eventfd(0, EFD_NONBLOCK | EFD_SEMAPHORE);
```

Comment:
After looking at the man page, I think it's buggy to use close.
If a file descriptor being monitored by select() is closed in another thread, the result is unspecified. On some UNIX systems, select() unblocks and returns, with an indication that the file descriptor is ready (a subsequent I/O operation will likely fail with an error, unless another the file descriptor reopened between the time select() returned and the I/O operations was performed). On Linux (and some other systems), closing the file descriptor in another thread has no effect on select(). In summary, any application that relies on a particular behavior in this scenario must be considered buggy. 

sorry about this. I should have studied it before shouting out loudly.